src/main/java/seedu/address/model/person/lab/LabList.java

```diff
@@ -0,0 +1,134 @@
+
+    /**
+     * Returns true if the list contains an equivalent lab as the given argument.
+     *
+     * @param toCheck The Lab to be checked
+     */
+    public boolean contains(Lab toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameLab);
+    }
+
+    /**
+     * Adds a lab to the list.
+     * The person must not already exist in the list.
+     *
+     * @param toAdd The Lab that you want to add.
+     */
+    public void add(Lab toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateLabException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the lab {@code target} in the list with {@code editedLab}.
+     * {@code target} must exist in the list.
+     * The lab identity of {@code editedLab} must not be the same as another existing lab in the list.
+     */
+    public void setLab(Lab target, Lab editedLab) {
+        requireAllNonNull(target, editedLab);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new LabNotFoundException(target.labNumber);
+        }
+
+        if (!target.equals(editedLab) && contains(editedLab)) {
+            throw new DuplicateLabException();
+        }
+
+        internalList.set(index, editedLab);
+    }
+
+    /**
+     * Removes the equivalent lab from the list.
+     * The lab must exist in the list.
+     *
+     * @param toRemove Lab you want to remove from the list.
+     */
+    public void remove(Lab toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new LabNotFoundException(toRemove.labNumber);
+        }
+    }
+
+    public void setLabs(LabList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code labs}.
+     * {@code labs} must not contain duplicate labs.
+     */
+    public void setLabs(List<Lab> labs) {
+        requireAllNonNull(labs);
+        if (!labsAreUnique(labs)) {
+            throw new DuplicateLabException();
+        }
+
+        internalList.setAll(labs);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Lab> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<Lab> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof LabList // instanceof handles nulls
+                && internalList.equals(((LabList) other).internalList));
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    /**
```

Comment:
Was wondering what's the difference between the classes `LabList` and `MasterLabList`? because both seem to have similar methods

Reply:
To answer your question on what's the difference between `MasterLabList` and `LabList`, there really isn't much of a difference, just that during our project discussions we noted that we might want to have a master list that stores all the current labs added to the system so when a new student arrives, we can auto-add those in for the new student. I think as of right now we can indeed do away with the `MasterLabList` class and just use a `LabList` instance to act as the masterlist that we talked about, but I was worried that we might need to have different behavior for the 2 in the future that's why I added this class. Feel free to let me know if you think this would be unnecessary cause I do think that it is a valid point.